src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -359,6 +367,73 @@ def get_package_size(self, output):
+                image_default_patch_mode = self.env_layer.file_system.read_with_retry(self.image_default_patch_mode_file_path)
+                settings = image_default_patch_mode.strip().split('\n')
+                for patch_mode in settings:
+                    if self.update_package_list in str(patch_mode):
+                        self.update_package_list_value = re.search(self.update_package_list + ' *"(.*?)".', str(patch_mode)).group(1)
+                    if self.unattended_upgrade in str(patch_mode):
+                        self.unattended_upgrade_value = re.search(self.unattended_upgrade + ' *"(.*?)".', str(patch_mode)).group(1)
+
+                backup_image_default_patch_mode_json = {
+                    self.update_package_list: self.update_package_list_value,
+                    self.unattended_upgrade: self.unattended_upgrade_value
+                }
+                self.composite_logger.log_debug("Logging default system settings for auto OS updates. [Settings={0}] [Log file path= {1}]"
+                                                .format(str(backup_image_default_patch_mode_json), self.image_default_patch_mode_backup_path))
+                self.env_layer.file_system.write_with_retry(self.image_default_patch_mode_backup_path, '{0}'.format(json.dumps(backup_image_default_patch_mode_json)), mode='w+')
+        except Exception as error:
+            error_message = "Exception during fetching and logging default auto update settings on the machine. [Exception={0}]".format(repr(error))
+            self.composite_logger.log_error(error_message)
+            raise
+
+    def is_image_default_patch_mode_backup_valid(self, image_default_patch_mode_backup):
+        if self.update_package_list in image_default_patch_mode_backup or self.unattended_upgrade in image_default_patch_mode_backup:
+            return True
+        return False
+
+    def update_image_default_patch_mode(self, patch_mode, value="0"):
+        """ Updates (or adds if it doesn't exist) the given patch_mode with the given value in image_default_patch_mode_file """
+        try:
+            # todo: adding space between the patch_mode and value since, we will have to do that if we have to add a patch_mode that did not exist before
+            self.composite_logger.log("Updating default system settings for auto OS updates. [Patch Mode={0}. Value={1}]".format(str(patch_mode), value))
+            image_default_patch_mode = self.env_layer.file_system.read_with_retry(self.image_default_patch_mode_file_path)
+            patch_mode_to_update = patch_mode + ' "' + value + '";'
+            patch_mode_found_in_file = False
+            updated_patch_mode = ""
+            settings = image_default_patch_mode.strip().split('\n')
+            for i in range(len(settings)):
+                # ToDo: do we need a check for value as well, to check if the value is not the same as the one sent in to replace it, as that will be an unnecessary write
```

Comment:
Since it's a simple to do, please act on the to do prior to final iteration or just remove it. I'm okay with an unnecessary write given the trivial file size and complexity, and that there should be any increase in probability of consequential failure in such a write.

Reply:
removed the comment, keeping the extra write